### PR TITLE
Add setDomainSuffixMatch to WifiEnterpriseConfig for APi >=29

### PR DESCRIPTION
**Pull request by JeGoi**
Fix setWpa2EnterpriseConfig for android api >= 29

https://developer.android.com/reference/kotlin/android/net/wifi/WifiNetworkSuggestion.Builder#setWpa2EnterpriseConfig(android.net.wifi.WifiEnterpriseConfig)

https://cs.android.com/android/platform/superproject/+/master:frameworks/base/wifi/java/android/net/wifi/WifiEnterpriseConfig.java;l=1426;drc=master

### CODE REVIEW

File: src/main/java/org/packetfence/agent/MainActivity.java [hunk 1]
```diff
@@ -494,6 +495,14 @@ public void configureWirelessConnectionWPAPEAPAndEAPTLS(Object[] categoryObj, Ha
                         MainActivity.this.userP12Name = (String) config.get("PayloadDisplayName");
                         MainActivity.this.tlsUsername = (String) config.get("PayloadCertificateFileName");
                     }
+                    if (payloadType.equals("com.apple.security.pkcs1")) {
+                        showInDebug("Found the EAP-TLS root certificate");
+                        MainActivity.this.serverCN = (String) config.get("PayloadCertificateFileName");
+                    } else {
+                        if (MainActivity.this.api_version >= 29) {
+                            showInBox("Your PacketFence server needs a patch or change manually the html/captive-portal/templates/wireless-profile-tls.xml and allow access to 'com.apple.security.pkcs1'");
```
Comment (extrafu) [inline src/main/java/org/packetfence/agent/MainActivity.java]: I think telling that to end-users is 'useless' - better just say "Your Android version..... blabla. Please contact your system administrator."

Comment (JeGoi) [reply src/main/java/org/packetfence/agent/MainActivity.java]: OK

**Comment by JeGoi** [comment]
Need to modify/update/upgrade:
https://github.com/inverse-inc/packetfence/blob/devel/html/captive-portal/templates/wireless-profile-tls.xml
https://github.com/inverse-inc/packetfence/blob/devel/html/captive-portal/templates/wireless-profile-peap.xml

Missing:
- [x] Exit if API>=29 and no com.apple.security.pkcs1
- [x] Need test on api 29 (android 10)
- [x] Need a fix for eap peap

**Comment by JeGoi** [comment]
Need to wait according to #https://github.com/inverse-inc/packetfence/issues/5966
This will change the way eap peap is parsing the xml file to extract information.

**Review by julsemaan — APPROVED** [review]

**Pull request merged**